providers/implementations/ciphers/ciphercommon_block.c

```diff
@@ -105,12 +106,21 @@ void ossl_cipher_padblock(unsigned char *buf, size_t *buflen, size_t blocksize)
         buf[i] = pad;
 }
 
+/*
+ * ossl_cipher_unpadblock removes the padding from the final block in constant time.
+ * Note that if the padding is invalid, ERR_raise is not called in order to preserve
+ * constant-timedness.
+ */
 int ossl_cipher_unpadblock(unsigned char *buf, size_t *buflen, size_t blocksize)
 {
     size_t pad, i;
     size_t len = *buflen;
+    size_t gp = 1;
```

Comment:
I'm still not a fan of the gp = 1. But it's at least in combination with a select function now, which should prevent the compiler from learning possible values.

Reply:
Any suggestions?

Reply:
We have constants for it here: https://github.com/openssl/openssl/blob/d68820d95634322108316f3051a1746ead88adaf/test/constant_time_test.c#L18-L27

They can probably be moved to the header file instead.

I suggest you create a variable initialized to TRUE, and at the end of the function do a return constant_time_select on it.